newIDE/app/src/CommandPanel/manager.js

```diff
@@ -0,0 +1,61 @@
+// @flow
+
+export type Command = {
+  name: string, // Unique identifier
+  text: string, // Display text
+  enabled: boolean,
+  handler: () => void | Promise<void>,
+};
+
+class CommandManager {
+  commands: Array<Command>;
```

Comment:
I see a lot of find/findIndex - which can becomes costly when we'll have 200 commands. Do you think we could store instead commands in an object (`{[string]: Command}`), indexed by the command name? This means that access will be done in constant time, whatever the number of commands.